### ISSUE TITLE
Global Search UI issues

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#8054]
https://github.com/JabRef/jabref/pull/8045 introduced global search. There are some open points:

- [x] resizing the global search window should not change the column width of the main table (maybe refs https://github.com/JabRef/jabref/pull/7181)
- [x] Library not shown "on my machine"\
    ![grafik](https://user-images.githubusercontent.com/1366654/131904365-db9851a2-db47-423b-8950-f4fc43221ac7.png)
- [ ] Selecting an entry in the global search should select the entry in the main table
- [x] (Optional) the places where the search hit should be highlighted\
  (we had this functionality in JabRef 2.9 beta - https://github.com/JabRef/jabref/blob/4e23c05838a3dd6e5f7870e804174a5ec043bc3c/CHANGELOG#L252). (Personal remark: implement in JabRef main branch 10 years ago - https://sourceforge.net/p/jabref/patches/147/#1ebb).

**Comment by mlep** [comment]
I am happy to see that the global search is back! Very useful feature!

A few glitches:
- [ ]  If global search is NOT activated, the button `Search across libraries window` still works for the current library (which is great!). So, I guess it would be better described as `Display results in an external window` (or something like this)
- [ ] Using dark theme: in the external window, the preview area is white if no entry is selected (then it becomes dark grey once an entry is selected). The UI would be more consistent if the preview area was dark grey from the beginning.
- [ ] In the external window, the button `Keep dialog always on top` is located in a weird place. Maybe it would be better if is was moved lower, on the same line as the button `OK`
- [ ] The UI does not retain the size of the Search across libraries window`. Open it, changes its size, close it, open it again : it has its original size.
- [ ] The `Search across libraries` window does not have a title.

I also came across a raised exception, but cannot reproduce it (yet)...

**Comment by Siedlerchr** [comment]
Thanks for the detailed feedback @mlep  Regarding the special fields to display @calixtus  and I decided not to show them, because it does make sense that you can modify them. Could also produce some side effects...

**Comment by Siedlerchr** [comment]
I found the issue with the library field. Totally stupid. As I previously tested the ugly hack, it modified the entries and added the library field. Normally the library field is a biblatex field for a different purpose.

I will look into a different way to get the library information

**Comment by Siedlerchr** [comment]
@koppor  
> (Optional) the places where the search hit should be highlighted
(we had this functionality in JabRef 2.9 beta -  

The functionality is also  implemented in the current versions since a long time in the Preview Area

**Comment by koppor** [comment]
- [ ] Changing column sizes of the global search window should NOT resize columns in the main table. Somehow related: https://github.com/JabRef/jabref/pull/7181

**Comment by Siedlerchr** [comment]
Current status after #8294 is merged

- Window has a title
- Search string is applied when opening a new library and the option "keep search string across libraries" is selected

Only issues left:
- Preview in dark theme when no search results is still "white"
- Location of the pin to top button 
- Debatable: If the columns like status or relevant should be shown, because you could theoretically change the value of them in the search window...
- When marking an entry in the external window, it should be marked in the corresponding library (e.g. select the library, select the item in the main table)

**Comment by koppor** [comment]
We opened a follow-up issue to be able to focus on the open points.
